static/js/compose.js

```diff
@@ -402,6 +405,37 @@ function compose_error(error_text, bad_input) {
     }
 }
 
+function send_typing_notification_ajax(recipients, operation) {
+    channel.post({
+        url: '/api/v1/typing',
+        data: {
+          to: recipients,
+          op: operation,
+        },
+        success: console.log,
+        error: function (xhr, error_type) {
+            var response = channel.xhr_error_message("Error sending typing notification", xhr);
+            console.log(response);
+        }
+    });
+}
+
+function send_typing_started_notification() {
+    function check_and_send() {
+        if (compose.message_content() && compose.recipient()) {
+            console.log("Going to send typing started notification to " + compose.recipient());
+            send_typing_notification_ajax(compose.recipient(), 'start');
+        }
+    }
+    check_and_send();
+    setInterval(check_and_send, 15000);
```

Comment:
Add it to JS_SPECS in settings.js and `check-all.js`;  as a sidenote, `$ git grep muting_ui` (or another small JS file) is how I checked what all the things you need to do when adding a new module are :).  And look at any small module to see how we export it.